linkerd/metrics/src/lib.rs

```diff
@@ -31,3 +31,28 @@ macro_rules! metrics {
         )+
     }
 }
+
+pub trait Factor {
+    fn factor(n: u64) -> f64;
+}
+
+pub struct MillisAsSeconds;
+
+/// Largest `u64` that can fit without loss of precision in `f64` (2^53).
+///
+/// Wrapping is based on the fact that Prometheus models values as f64 (52-bits
+/// mantissa), thus integer values over 2^53 are not guaranteed to be correctly
+/// exposed.
+const MAX_PRECISE_UINT64: u64 = 0x20_0000_0000_0000;
+
+impl Factor for () {
+    fn factor(n: u64) -> f64 {
+        n.wrapping_rem(MAX_PRECISE_UINT64 + 1) as f64
```

Comment:
i was kind of wondering if the wrapping module by the max precise value could be factored out so that `Factor` impls don't have to be aware of it...i suppose we could factor the max precise value too and do the modulo in `Counter::value`. 

this is probably not worth the effort.

Reply:
The intent for doing this inside of factor is so that we can adjust the precision based on the scale of the value:

https://github.com/linkerd/linkerd2-proxy/pull/754/files#diff-36a73a9a9de1e34abac85003560b0aea70e9ebb0f1cc29af3dce703bfbef7a89R56

Now that I look at this again, though, it's probably not really correct; since we may lose some of the precision before scaling it.

Reply:
Actually, the current behavior [seems fine](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=4004452323dc29389db7772f97f1a325)